### ISSUE TITLE
Take care of encoding of the read files (pylint W1514, PEP598)

### ISSUE THREAD

**Issue by mknorps** [tweag/FawltyDeps#21]
# General

Solve issue underlying to 
> W1514: Using open without explicitly specifying an encoding (unspecified-encoding)

The issue is raised after a discussion with Johan and his investigations.

# Problem

## Minimal example to reproduce

```
from pathlib import Path

def uninferable(path: Path):
    return path.read_text()

def inferable(path: Path):
    return Path(path).read_text()

print(uninferable(Path("/dev/null")))
print(inferable(Path("/dev/null")))
```

Pylint issue tracked: https://github.com/PyCQA/pylint/issues/4813 which indicates that this is not a simple bug.

## Explanation

Running pylint on this will generate W1514 for the second `.read_text()` call, but not for the first.

In the first function pylint seems unable to infer that path is for sure a `Path` object. It seems to ignore/not understand type annotations, and requires instead a call to the `Path(...)` constructor inside the same function to be sure about the type. When pylint is not sure about the type, it cannot claim that calling `.read_text()`  is definitely going to open a file, so it says nothing instead.

# Proposed solution

Promote a CPython version/configuration that treats everything as UTF-8 by default (e.g. by running python with `-X utf8` or `PYTHONUTF8=1` in the environment).

In other words, maybe we should add a check on startup to warn if the current locale/default encoding is NOT UTF-8, and then we can assume that `open()`  will do the Right Thing without passing encoding everywhere, and disable this warning.

**Comment by mknorps** [comment]
This is a pylint issue. We agreed that we want to change pylint to other linter anyway, so I close this ticket without resolving.
